File: src/main/java/com/example/homework/detail/DetailPresenterImpl.java
package com.example.homework.detail;

import android.support.annotation.NonNull;

import com.example.homework.data.Translation;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class DetailPresenterImpl implements DetailContract.DetailPresenter {
    private DetailContract.DetailUIView detailUIView;
    private DetailModel detailModel = new DetailModel();

    public DetailPresenterImpl(DetailContract.DetailUIView detailUIView) {
        this.detailUIView = detailUIView;
    }

    @Override
    public void initData(SecondActivity secondActivity) {
        detailModel.setIntent(secondActivity);
        Call<Translation> call = detailModel.update(detailModel.id);
        call.enqueue(new Callback<Translation>() {
            @Override
            public void onResponse(Call<Translation> call, Response<Translation> response) {
//                String responseData = response.body() != null ? response.body().toString() : null;
//                newsModel.parseJSONWithJSONObject(responseData);
                Translation translation = response.body();
                if (translation != null) {
                    detailUIView.initView(translation.getBody());
                }
            }

            @Override
            public void onFailure(@NonNull Call<Translation> call, @NonNull Throwable t) {
                detailUIView.onError();
                t.printStackTrace();
            }
        });
    }
}

File: main/java/com/example/homework/detail/DetailPresenterImpl.kt
package com.example.homework.detail

import com.example.homework.data.Translation

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class DetailPresenterImpl(private val detailUIView: DetailContract.DetailUIView) : DetailContract.DetailPresenter {
    private val detailModel = DetailModel()

    override fun initData(secondActivity: SecondActivity) {
        detailModel.setIntent(secondActivity)
        val call = detailModel.update(detailModel.id)
        call.enqueue(object : Callback<Translation> {
            override fun onResponse(call: Call<Translation>, response: Response<Translation>) {
                //                String responseData = response.body() != null ? response.body().toString() : null;
                //                newsModel.parseJSONWithJSONObject(responseData);
                val translation = response.body()
                if (translation != null) {
                    detailUIView.initView(translation.body!!)
                }
            }

            override fun onFailure(call: Call<Translation>, t: Throwable) {
                detailUIView.onError()
                t.printStackTrace()
            }
        })
    }
}
